compiler/arm/codegen/OMRInstruction.cpp

```diff
@@ -142,7 +142,7 @@ OMR::ARM::Instruction::remove()
 
 void OMR::ARM::Instruction::ARMNeedsGCMap(uint32_t mask)
    {
-   if (TR::comp()->useRegisterMaps())
+   if (self()->cg()->comp()->useRegisterMaps())
```

Comment:
The `CodeGenerator` object is not cached in the `Instruction` so what you have won't work.

I think the `Compilation` object should be passed in as a parameter to `ARMNeedsGCMap` instead.  i.e.,
```
void OMR::ARM::Instruction::ARMNeedsGCMap(uint32_t mask, TR::Compilation *comp)
```
The overhead of passing an extra parm should be much less than the TLS lookup that the `TR::comp` was incurring.

Reply:
Did you mean something like that? (I force-pushed the prev commit).

Reply:
Hm...  Now that I see your code, most of the places that seem to use this API have a CodeGenerator object in hand already and have to reach for the Compilation object.  Because this is arguably an API used in the CodeGenerator, perhaps you should pass a `TR::CodeGenerator *` pointer instead, and then within this function indirect it to get the `TR::Compilation *` as needed?  i.e.,
```
void OMR::ARM::Instruction::ARMNeedsGCMap(uint32_t mask, TR::CodeGenerator *cg)
```
You can dig the Compilation pointer out by using `cg->comp()`.

Does that make sense?